packer/build.go

```diff
@@ -52,6 +52,46 @@ type CoreBuild struct {
 	prepareCalled bool
 }
 
+type BuildMetadata struct {
+	PackerVersion string
+	Plugins       map[string]PluginDetails
+}
+
+func (b *CoreBuild) getPluginsMetadata() map[string]PluginDetails {
+	resp := map[string]PluginDetails{}
+
+	builderPlugin, builderPluginOk := PluginsDetailsStorage[fmt.Sprintf("%q-%q", PluginComponentBuilder, b.BuilderType)]
+	if builderPluginOk {
+		resp[builderPlugin.Name] = builderPlugin
+	}
+
+	for _, pp := range b.PostProcessors {
+		for _, p := range pp {
+			postprocessorsPlugin, postprocessorsPluginOk := PluginsDetailsStorage[fmt.Sprintf("%q-%q", PluginComponentPostProcessor, p.PType)]
+			if postprocessorsPluginOk {
+				resp[postprocessorsPlugin.Name] = postprocessorsPlugin
+			}
+		}
+	}
+
+	for _, pv := range b.Provisioners {
+		provisionerPlugin, provisionerPluginOk := PluginsDetailsStorage[fmt.Sprintf("%q-%q", PluginComponentProvisioner, pv.PType)]
+		if provisionerPluginOk {
+			resp[provisionerPlugin.Name] = provisionerPlugin
+		}
+	}
+
+	return resp
+}
+
+func (b *CoreBuild) GetMetadata() BuildMetadata {
```

Comment:
I think it'd be good to unit test this method, I think the tests could be added to https://github.com/hashicorp/packer/blob/main/packer/build_test.go this file, and just call GetMetadata after setting up a Core Build object that would have metadata, and just check that the expected model is populated